### DISCUSSION TITLE
Manipulating scores in multisite

### DISCUSSION THREAD

**Discussion by JulieVelghe** [craftcms/cms#13801] [Q&A] [answered]
I'm working on a searching module where it would be possible to boost the scores set by a config file. For example:
    'events' => [
        'sectionHandle' => 'events',
        'hasTemplate' => true,
        'fieldsWeights' => [
            'title' => 5,
            'contentBlocks' => 2,
        ],
        'searchWeight' => 1,
    ],

Where the searchWeight is a boost if the 'searchresult' is of that section type, and an individual field boost.

The problem occurs when I'm searching in a multisite. 
I'm using the EVENT_AFTER_SEARCH event. The $e->results gives me correct data from all sites. However, the scores are grouped by the element id, without any consideration of the site id. In my results, I do have some elements with the same elementID but with a different site id.  Therefore, score-manipulating based on the elementId isn't truly correct.

An example as to why this could be helpfull:
let's say we have a site for private individuals and for firms. We have a page about topic, that is active for both sites, but on the firm-side, the content is more elaborate than on the private individuals side.  Therefore, I would assume that there should be a score for both the sites, and that that score could differ, even though it's the same element.

Anyone has any idea how I could accomplish this? Or is this already on the roadmap in some way?

**Comment by brandonkelly** [comment]
Looks like this is something we overlooked when adding support for multi-site element queries.

Most element queries only fetch elements in a single site at a time, in which case `SearchEvent::$siteId` will be set to a single integer. In those cases, the scores will only be based on the keyword matches for that one site, and everything should be working as expected.

The problem is that the `Search` service really should be grouping scores by element ID _and_ site ID, so elements don’t end up with additional points in the scoring when multiple of their sites are queried at once.

Unfortunately there wouldn’t be a non-breaking way to fix this in Craft 4, so we will have to address it in Craft 5 instead.

**Comment by brandonkelly** [comment, marked as answer]
This is now resolved for Craft 5, via 5d8ac94f0972e2662b52199922c0ae9c19d3a596.
